src/osd/scheduler/mClockScheduler.cc

```diff
@@ -383,31 +376,18 @@ int mClockScheduler::calc_scaled_cost(op_type_t op_type, int cost)
 bool mClockScheduler::maybe_update_client_cost_info(
   op_type_t op_type, int new_cost)
 {
-  int capped_item_cost = 4 * 1024 * 1024;
-
   if (new_cost == 0) {
     return false;
   }
 
-  // The mclock params represented in terms of the per-osd capacity
-  // are scaled up or down according to the cost associated with
-  // item cost and updated within the dmclock server.
+  // Calculate the new cost if the new item cost for the
+  // op_type is different from the current cost.
   int cur_cost = client_cost_infos[op_type];
-
-  // Note: Cap the scaling of item cost to ~4MiB as the tag increments
-  // beyond this point are too long causing performance issues. This may
-  // need to be in place until benchmark data is available or a better
-  // scaling model can be put in place. This is a TODO.
-  if (new_cost >= capped_item_cost) {
-    new_cost = capped_item_cost;
-  }
-
   bool cost_changed =
     ((new_cost >= (cur_cost << 1)) || (cur_cost >= (new_cost << 1)));
```

Comment:
Does this caching have any effect on performance? If not, let's get rid of it - it makes it much simpler to understand.

With a range of costs treated the same, the actual cost is dependent on the I/O sequence and could have unintended effects if e.g. the first i/o is 51% the size of subsequent I/Os

Reply:
I have removed the caching as per the previous comment.